RxJava 수정

pengin7384 opened the pull request:
## 개요
- 기존에 적용하였던 RxJava의 잘못된 부분을 찾고 해당 부분을 고치기


## 구현사항
- searchAddress 흐름 변경
- 에러메시지 처리 변경


## 리뷰 요청사항
- RxJava 처리가 잘못된 부분을 집중적으로 리뷰 받고 싶습니다.


resolve:#35

HoneyPot8014 committed:
feature/U-019

do UI work myPageFragment

HoneyPot8014 committed:
feature/U-019

update fragment_my_page UI.
delete change user info buttons from collapsingLayout, move it to bottom buttons

HoneyPot8014 committed:
feature/U-019

update UI when user sign in.
change signInFragment to MyPageFragment when user sign in
in MyPage, can see their information that from remote user data

HoneyPot8014 committed:
feature/U-019

add data function. remote Repository query
before needs to change nickName, check the nickName is existed.

HoneyPot8014 committed:
feature/U-019

add change nick name dialog view
in MyPageFramgnet, when click change nickname button, this dialog pop up

HoneyPot8014 committed:
feature/U-019

add change password dialog view
in MyPageFramgnet, when click change password button, this dialog pop up

HoneyPot8014 committed:
feature/U-019

add function to check nickname is existed.
query database by nickname. when existed fire a exception
to do : not existed duplicate nickName, can change user's nickName

sabgilhun committed:
add address searcher

HoneyPot8014 committed:
feature/U-019

add function to update password
this function is available only sign in with email and password
when sign in with google or facebook, this function is not appeared in UI

pengin7384 committed:
Error Message를 Constants에 정의 후 사용, getAddressList 람다식 변환

pengin7384 committed:
Error Message를 Constants에 정의 후 사용, getAddressList 람다식 변환

HoneyPot8014 committed:
feature/U-019

add function to update profile
start intent for pick a picture, retrieve from its uri.
then, save file binary to remote storage, and update database.

pengin7384 committed:
searchAddress 흐름 변경

기존의 뷰에서 mAddressList를 observe하던 구조를 삭제, searchAddress를 호출시 처리하는 형태로 변경. ViewModel에서는 List<TMapPOIItem>을 List<Address>로 변환하여 View에게 전달.

pengin7384 committed:
searchAddress 흐름 변경

기존의 뷰에서 mAddressList를 observe하던 구조를 삭제, searchAddress를 호출시 처리하는 형태로 변경. ViewModel에서는 List<TMapPOIItem>을 List<Address>로 변환하여 View에게 전달.

ParkSangGwon commented on app/src/main/java/com/swsnack/catchhouse/data/roomsdata/remote/RoomsRemoteData.java:
```diff
@@ -25,19 +26,14 @@ private RoomsRemoteData() {
-                } else {
-                    subscribe.onError(new RuntimeException("Can't FindAllPOI(No Data)"));
-                }
+        return Single.defer(() -> Single.create(subscribe -> mTMapData.findAllPOI(keyword, arrayList -> {
```
`Single.defer()`는 없어도 됩니다.

pengin7384 replied:
반영하였습니다!

ParkSangGwon commented on app/src/main/java/com/swsnack/catchhouse/view/fragments/MapFragment.java:
```diff
@@ -68,7 +72,7 @@ public void onViewCreated(@NonNull View view, @android.support.annotation.Nullab
         AddressBindingAdapter adapter = new AddressBindingAdapter(getContext());
         getBinding().rvMapAddress.setAdapter(adapter);
-        adapter.updateItems(getViewModel().getAddressList().getValue());
+        //adapter.updateItems(getViewModel().getAddressList().getValue());
```
안쓰는 함수는 지워주세요

pengin7384 replied:
넵 정리하도록 하겠습니다

pengin7384 committed:
String이었던 address를 Address로 변환

ParkSangGwon commented on app/src/main/java/com/swsnack/catchhouse/view/fragments/MapFragment.java:
```diff
@@ -78,26 +82,25 @@ public void onViewCreated(@NonNull View view, @android.support.annotation.Nullab
-
-        getViewModel().getAddressList().observe(this, addressObserver);
-
         getBinding().etMapSearch.setOnKeyListener((v, keyCode, event) -> {
```
여기서 작성된 코드는 아니지만 `setOnEditorActionListener()`를 사용하는게 더 좋습니다.
예제를 찾아보시면 사용방식은 금방 아실 수 있을거에요

pengin7384 replied:
감사합니다. 관련 정보를 찾아보고 반영하였습니다!

ParkSangGwon commented on app/src/main/java/com/swsnack/catchhouse/view/fragments/MapFragment.java:
```diff
@@ -78,26 +82,25 @@ public void onViewCreated(@NonNull View view, @android.support.annotation.Nullab
             if ((event.getAction() == KeyEvent.ACTION_DOWN) && (keyCode == KeyEvent.KEYCODE_ENTER)) {
-                getViewModel().searchAddress();
-                getBinding().rvMapAddress.setVisibility(View.VISIBLE);
+                getViewModel().searchAddress()
```
- RxJava에서는 disposable에 대한 관리가 필요합니다.
- 구독중인데 Fragment가 종료되어도 계속 구독중이므로 Disposable에 대한 `dispose()`처리가 필요합니다.
- activity/fragment가 destroy되는 시점에 disposable들도 처리해주어야 합니다.

pengin7384 replied:
CompositeDisposable를 통해 처리하도록 추가하였습니다!

pengin7384 committed:
Merge pull request #38 from boostcampth/feature/S-037

Room 변경

ParkSangGwon commented on app/src/main/java/com/swsnack/catchhouse/viewmodel/searchviewmodel/SearchViewModel.java:
```diff
@@ -18,66 +18,51 @@
-        Log.v("csh", "key:"+getKeyword());
+                .doOnSubscribe(__ -> mListener.isWorking())
+                .doAfterTerminate(() -> mListener.isFinished())
+                .map(tMapPOIItems -> {
```
-  `toObservable()`, `flatMap()`, `map()`을 사용해서 더 간단명료하게 줄일 수 있습니다.
```java
return mRepository.getPOIFromRemote(getKeyword())
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .doOnSubscribe(__ -> mListener.isWorking())
                .doAfterTerminate(() -> mListener.isFinished())
                .toObservable()
                .flatMap(Observable::fromIterable)
                .map(item -> new Address(item.name, item.getPOIAddress().replace("null", ""), item.getPOIPoint().getLongitude(), item.getPOIPoint().getLatitude()))
                .toList();
```

lunar999 replied:
튜터링 시간이나 슬렉채널에서도 이야기했었는데요  함수형 프로그래밍을 하려면 아예 사고방식을 바꿔야 한다고  했었는데요 상권님이 잘 알려주셨네요 보통 람다식안에 지역변수, for 문, try, catch를 거의 사용하지 않습니다 참고하세요.

pengin7384 replied:
코드가 훨씬 더 간결해진거 같습니다! 감사합니다

pengin7384 replied:
@lunar999 감사합니다! 앞으로 람다식안에 지역변수, for문, try문을 자제하도록 하겠습니다!

HoneyPot8014 committed:
feature/U-019

fix storage bugs
when user delete from firebase, it has occured an storageError when user don't save profile in firebase Storage.
and divide viewModel function from Model.

HoneyPot8014 committed:
Merge branch 'develop' into feature/U-019

HoneyPot8014 committed:
Merge pull request #40 from boostcampth/feature/U-019

마이 페이지 개발

ParkSangGwon committed:
review-190206 adapter 리뷰

ParkSangGwon committed:
review-190206 data 패키지 리뷰

ParkSangGwon committed:
review-190206 view 패키지 리뷰

HoneyPot8014 committed:
Merge pull request #42 from boostcampth/review/190206

Review/190206

HoneyPot8014 committed:
feature/U-019

BaseFragment의 setViewModel 함수 이름 수정

HoneyPot8014 committed:
feature/43

SignUpFragment의 라디오 버튼단위의 리스너를 라디오 그룹으로 변경

HoneyPot8014 committed:
feature/43

MyPageFragment의 로그인 방법에 따라, 비밀번호 수정항목을 보여주는 로직 변경.
로그인한 방법이 여러개인 경우, list로 로그인 이력을 가져오기 때문에, 특정 index를 가져오지 않고, list를 순회하며 조회

HoneyPot8014 committed:
feature/43

MyPageFragment의 프로필 이미지 변경 중, 뒤로가기 버튼을 누를 시, 원하지 않던 snack 메세지 노출 수정

HoneyPot8014 committed:
feature/43

View - ViewModel 의 에러처리 방식 변화
기존에, ViewModel에서 Throwable을 던져 View에서 스낵바를 띄우던 형태에서, ViewModel이 에러를 처리하고, View에서 에러 메세지를 스낵바로 띄우는 형태로 변화

HoneyPot8014 committed:
feature/43

Application 클래스 생성. context 접근하도록 singleTon으로 구현. Multidex 지원

sabgilhun committed:
refactoring

HoneyPot8014 committed:
feature/43

BottomNavigation의 fragment 변화를 viewPager로 변경
기존의 frameLayout의 replace 대신, viewpager의 프래그먼트를 전환하는 걸로 변경.

HoneyPot8014 committed:
feature/43

SignUpFragment 회원가입 프로필 사진 선택 중, 취소 시 의도하지 않은 스낵바 노출 제거

HoneyPot8014 committed:
feature/43

User POJO setter 변경

HoneyPot8014 committed:
feature/43

data패키지 profile 함수 파라미터 변경 (byte[] -> uri)

HoneyPot8014 committed:
feature/43

jenkins 오류 해결 커밋.

HoneyPot8014 committed:
Merge pull request #45 from boostcampth/feature/43

FIXME 반영 수정. Jenkins 빌드 오류 해결

HoneyPot8014 committed:
feature/C-46

ChattingFragment, ViewModel 추가

HoneyPot8014 committed:
feature/C-46

chattingList UI 작업

HoneyPot8014 committed:
feature/C-46

chattingList 인터페이스 작성

HoneyPot8014 committed:
feature/C-46

chattingList Adapter 개발

HoneyPot8014 committed:
feature/C-46

ChattingList Remote Database 일부 구현
채팅방 중복검사를 위한, 채팅방 유저들과의 중복 채팅방 조회 기능
채팅방 개설 기능
채팅방 리스트 검색 기능

HoneyPot8014 committed:
feature/C-46

ChattingList 아이템에 유저 정보 입력
프로필 사진과 유저 닉네임을 리싸이클러뷰 아이템에 노출

HoneyPot8014 committed:
Merge pull request #47 from boostcampth/feature/C-46

채팅 리스트 DB 설계 및, UI 작업

HoneyPot8014 committed:
feature/C-48

ChattingList 버그 수정
나의 채팅 룸 뿐만 아니라, 다른 채팅룸까지 조회하는 버그 수정.

HoneyPot8014 committed:
feature/C-48

data 패키지 callback 구조 개선 리팩토링

HoneyPot8014 committed:
feature/C-48

로그인 관련 부분 오류 수정
Google, Facebook 로그인 후, 프로필사진, 닉네임 변경이 로그아웃 후 적용되지 않는 버그 수정

HoneyPot8014 committed:
feature/C-48

ChatListFragment 의 refresh 기능 추가 및, 함수 리팩토링
ChattingList를 한번 받아온 후, 프래그먼트 전환이 일어났을 때, update 되지 못한 오류 수정

HoneyPot8014 committed:
feature/C-48

ChattingListFragment의 갱신 효율 향상
로그아웃 상태여서도 chattingList를 요청하던 로직 변경

sabgilhun committed:
add function to upload room data

HoneyPot8014 committed:
Merge pull request #49 from boostcampth/feature/C-48

ChattingListFragment 오류 수정 및, 함수 리팩토링

sabgilhun committed:
pre merge

sabgilhun committed:
Merge remote-tracking branch 'origin/develop' into feature/S-034

# Conflicts:
#	app/src/main/java/com/swsnack/catchhouse/adapters/BaseDiffUtil.java
#	app/src/main/java/com/swsnack/catchhouse/adapters/bindingadapters/RoomsDataBinding.java
#	app/src/main/java/com/swsnack/catchhouse/adapters/chattingadapter/ChattingListAdapter.java
#	app/src/main/java/com/swsnack/catchhouse/constants/Constants.java
#	app/src/main/java/com/swsnack/catchhouse/data/AppDataManager.java
#	app/src/main/java/com/swsnack/catchhouse/data/chattingdata/remote/RemoteChattingManager.java
#	app/src/main/java/com/swsnack/catchhouse/data/userdata/APIManager.java
#	app/src/main/java/com/swsnack/catchhouse/data/userdata/UserDataManager.java
#	app/src/main/java/com/swsnack/catchhouse/data/userdata/api/AppAPIManager.java
#	app/src/main/java/com/swsnack/catchhouse/data/userdata/remote/AppUserDataManager.java
#	app/src/main/java/com/swsnack/catchhouse/view/activities/BottomNavActivity.java
#	app/src/main/java/com/swsnack/catchhouse/view/activities/WriteActivity.java
#	app/src/main/java/com/swsnack/catchhouse/view/fragments/ChatListFragment.java
#	app/src/main/java/com/swsnack/catchhouse/viewmodel/chattingviewmodel/ChattingViewModel.java
#	app/src/main/java/com/swsnack/catchhouse/viewmodel/roomsviewmodel/RoomsViewModel.java
#	app/src/main/java/com/swsnack/catchhouse/viewmodel/userviewmodel/UserViewModel.java
#	app/src/main/res/values/styles.xml

sabgilhun committed:
modify wrong merge

sabgilhun committed:
apply PR

sabgilhun committed:
Merge pull request #51 from boostcampth/feature/S-034

판매자가 매물 정보를 올리는 기능

sabgilhun committed:
HomeFragemnt extends BaseFragment

sabgilhun committed:
delete OnSearchButtonListener

sabgilhun committed:
Merge pull request #53 from boostcampth/feature/H-052

FIXME 수정

pengin7384 committed:
item 사이즈 변경

pengin7384 committed:
피드백 반영

안쓰는 코드 정리, setOnEditorActionListener로 수정, disposable 관리, rx 수정

pengin7384 committed:
Merge remote-tracking branch 'origin/feature/S-035' into feature/S-035

# Conflicts:
#	app/src/main/java/com/swsnack/catchhouse/data/roomsdata/remote/RoomsRemoteData.java
#	app/src/main/java/com/swsnack/catchhouse/view/fragments/MapFragment.java
#	app/src/main/java/com/swsnack/catchhouse/viewmodel/searchviewmodel/SearchViewModel.java
